include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -471,6 +471,88 @@ def NVMMASharedEncodingAttr :
   let hasCustomAssemblyFormat = 1;
 }
 
+def SwizzledBlocksSharedEncodingAttr :
```

Comment:
I kind of think we can have a better name for this new layout.
`SwizzledBlocks` makes me think it is swizzling blocks of elements. The original idea about this layout is that in the 1st maxPhase rows (1st block), we have phases: 0, 1, ..., maxPhase-1; in the 2nd block, we have phases: 1, 2, ..., maxPhase-1, 0; so on so forth. So I am thinking we should include something like "rotate" or "shift" in the name. And maybe also include "AMD" since we don't expect this layout to be used by nv path.